frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -348,6 +351,46 @@ export const getTopologyGroupItems = (dc: K8sResourceKind, groups: Group[]): Gro
   return groups;
 };
 
+export const getFilteredTrafficWorkload = (nodes: KialiNode[]): KialiNode[] =>
+  nodes.filter(({ data }) => data.nodeType === 'workload');
+
+export const getTrafficConnectors = (
+  trafficData: TrafficData,
+  resources: K8sResourceKind[],
+  consoleLinks: K8sResourceKind[],
+): Edge[] => {
+  const filteredWorkload = getFilteredTrafficWorkload(trafficData.nodes);
+  return trafficData.edges.reduce((acc, { data }) => {
+    const { data: sourceTrafficNode } = filteredWorkload.find(
+      (wrkld) => wrkld.data.id === data.source,
+    );
+    const { data: targetTrafficNode } = filteredWorkload.find(
+      (wrkld) => wrkld.data.id === data.target,
+    );
+    const sourceResourceNode = resources.find((res) => {
+      return res.metadata.name === sourceTrafficNode[sourceTrafficNode.nodeType];
+    });
+    const targetResourceNode = resources.find(
+      (res) => res.metadata.name === targetTrafficNode[targetTrafficNode.nodeType],
+    );
+    const kialiLink = consoleLinks?.find(
+      (cl) => cl.metadata.name === `kiali-namespace-${sourceResourceNode?.metadata?.namespace}`,
+    )?.spec?.href;
```

Comment:
There will be one kiali link per namespace, we can avoid finding it multiple times.